bot/exts/utils/clean.py

```diff
@@ -36,15 +41,80 @@ def mod_log(self) -> ModLog:
         """Get currently loaded ModLog cog instance."""
         return self.bot.get_cog("ModLog")
 
+    async def _delete_messages_individually(self, messages: List[Message]) -> None:
+        for message in messages:
+            # Ensure that deletion was not canceled
+            if not self.cleaning:
+                return
+            try:
+                await message.delete()
+            except NotFound:
+                # Message doesn't exist or was already deleted
+                continue
+
+    def _get_messages_from_cache(self, amount: int, predicate: CheckHint) -> Tuple[DefaultDict, List[int]]:
+        """Helper function for getting messages from the cache."""
+        message_mappings = defaultdict(lambda: [])
+        message_ids = []
+        for message in self.bot.cached_messages:
+            if not self.cleaning:
+                # Cleaning was canceled
+                return (message_mappings, message_ids)
+
+            if predicate(message):
+                message_mappings[message.channel].append(message)
+                message_ids.append(message.id)
+
+                if len(message_ids) == amount:
+                    # We've got the requested amount of messages
+                    return message_mappings, message_ids
+
+        # Amount exceeds amount of messages matching the check
+        return message_mappings, message_ids
+
+    async def _get_messages_from_channels(
+        self,
+        amount: int,
+        channels: Iterable[TextChannel],
+        predicate: CheckHint,
+        until_message: Optional[Message] = None
+    ) -> DefaultDict:
+        message_mappings = defaultdict(lambda: [])
+        message_ids = []
+
+        for channel in channels:
+
+            async for message in channel.history(amount=amount):
```

Comment:
```suggestion
            async for message in channel.history(limit=amount):
```
`amount` isn't a valid kwarg, but `limit` is.

Reply:
Am not sure why did I miss that, thanks for the sharp eye! https://github.com/python-discord/bot/pull/1527/commits/5bf42c7d3c4927dae2a130a95d83295db746338d